nicegui/run.py

```diff
@@ -72,21 +82,31 @@ async def cpu_bound(callback: Callable[P, R], *args: P.args, **kwargs: P.kwargs)
     It is encouraged to create static methods (or free functions) which get all the data as simple parameters (eg. no class/ui logic)
     and return the result (instead of writing it in class properties or global variables).
     """
+    if process_pool is None:
+        raise RuntimeError('Process pool not set up. Call run.setup() first.')
+
     return await _run(process_pool, safe_callback, callback, *args, **kwargs)
 
 
 async def io_bound(callback: Callable[P, R], *args: P.args, **kwargs: P.kwargs) -> R:
     """Run an I/O-bound function in a separate thread."""
+    if thread_pool is None:
+        raise RuntimeError('Thread pool not set up. Call run.setup() first.')
+
     return await _run(thread_pool, callback, *args, **kwargs)
 
 
 def tear_down() -> None:
     """Kill all processes and threads."""
     if helpers.is_pytest():
         return
-    assert process_pool is not None
-    for p in process_pool._processes.values():  # pylint: disable=protected-access
-        p.kill()
+
     kwargs = {'cancel_futures': True} if sys.version_info >= (3, 9) else {}
-    process_pool.shutdown(wait=True, **kwargs)
-    thread_pool.shutdown(wait=False, **kwargs)
+
+    if process_pool is not None:
+        for p in process_pool._processes.values():  # pylint: disable=protected-access
+            p.kill()
+        process_pool.shutdown(wait=True, **kwargs)
+
+    if thread_pool is not None:
+        thread_pool.shutdown(wait=False, **kwargs)
```

Comment:
```suggestion
    if process_pool is not None:
        for p in process_pool._processes.values():  # pylint: disable=protected-access
            p.kill()
        process_pool.shutdown(wait=True, **kwargs)
    if thread_pool is not None:
        thread_pool.shutdown(wait=False, **kwargs)
```

Blank lines should [only be used sparingly](https://legacy.python.org/dev/peps/pep-0008/#blank-lines:~:text=Use%20blank%20lines%20in%20functions%2C%20sparingly%2C%20to%20indicate%20logical%20sections.).